### PR TITLE
feat: Add comprehensive input validation with Zod schemas and database constraints

### LINKED ISSUE #45: Missing Input Validation
**Severity**: Medium  
**Category**: Data Integrity  

**Problem**:
Limited validation before database writes:

1. No validation that `startTime < endTime` for episodes
2. No validation of medication dosage ranges
3. No sanitization of user input strings
4. No validation of date formats

**Impact**:
- Invalid data can be stored in database
- Potential for data corruption
- App crashes from unexpected data
- Poor data quality for analytics

**Recommendations**:
1. Create validation schemas using `zod` or `yup`
2. Add validation layer in repositories before writes
3. Implement form validation in UI
4. Add constraints in database schema where possible
5. Create validation utility functions

**Files Affected**:
- `app/src/database/*.ts` (all repositories)
- Add new: `app/src/utils/validation.ts`
- Add new: `app/src/schemas/*.ts`

### PR DESCRIPTION
**Pull request by vfilby**
## Summary
Implements comprehensive input validation for all database writes using Zod validation schemas and database CHECK constraints. This addresses issue #45 by preventing invalid data from being persisted to the database through multiple layers of validation.

#### **src/schemas/episode.schema.ts**
Validates episodes, intensity readings, symptoms, pain locations:
- Episode startTime < endTime validation
- Intensity 0-10 range validation
- GPS coordinates within valid ranges (-90 to 90 lat, -180 to 180 long)
- Average intensity ≤ peak intensity
- String length limits (notes ≤ 5000 chars)
- Resolution time > onset time for symptoms

#### **src/schemas/medication.schema.ts**
Validates medications, doses, schedules, reminders:
- Positive dosage validation
- Time format validation (HH:mm 24-hour)
- Date range validation (startDate < endDate)
- Preventative medications require schedule frequency
- **Dose amount validation**: allows 0 for skipped doses, requires > 0 for taken doses
- Effectiveness rating 0-10
- Time to relief ≤ 1440 minutes (24 hours)
- Completed reminders must have completed_at timestamp

#### **src/schemas/dailyStatus.schema.ts**
Validates daily status logs:
- Date format (YYYY-MM-DD)
- No future dates allowed
- Status type only for yellow days

### 2. Repository Updates
Added validation to all repository create methods using `safeParse()`:
- **episodeRepository.ts** - Episode, intensity reading, symptom log, pain location log, episode note creation
- **medicationRepository.ts** - Medication, dose, schedule creation
- **dailyStatusRepository.ts** - Daily status log creation

Validation approach:
- Uses `safeParse()` for non-throwing error handling
- Logs validation errors in development mode
- Throws descriptive errors to prevent invalid data writes

### 3. Database Constraints (Schema v7 → v8)
Added CHECK constraints to enforce critical validations at database level:

**Episodes:**
- `start_time > 0`, `end_time > start_time`
- Intensity values 0-10, `average_intensity <= peak_intensity`
- Notes ≤ 5000 characters

**Intensity Readings:**
- `timestamp > 0`, `intensity` 0-10

**Symptom Logs:**
- `onset_time > 0`, `resolution_time > onset_time`
- Severity 0-10

**Medications:**
- Name 1-200 chars, type IN ('preventative', 'rescue')
- Dosage amount > 0, dosage unit 1-50 chars
- `end_date > start_date`
- `schedule_frequency` IN ('daily', 'monthly', 'quarterly')

**Medication Schedules:**
- Time format HH:mm (24-hour) using GLOB pattern
- Dosage > 0

**Medication Doses:**
- `amount >= 0`, status IN ('taken', 'skipped')
- **Business rule**: `status != 'taken' OR amount > 0` (taken doses must have positive amount)
- Effectiveness rating 0-10
- Time to relief ≤ 1440 minutes

**Medication Reminders:**
- Completed reminders must have `completed_at` timestamp
- `snoozed_until > scheduled_time`

**Daily Status Logs:**
- Date format YYYY-MM-DD (GLOB pattern)
- Status IN ('green', 'yellow', 'red')
- **Business rule**: `status = 'yellow' OR status_type IS NULL` (only yellow days can have status_type)

### 4. Test Fixes
- Fixed NewEpisodeScreen test by mocking `intensityRepository` (direct import, not through store)
- Fixed medication dose validation to allow `amount: 0` for skipped doses
- All 652 unit tests passing ✅
- All 15 E2E tests passing ✅

### Episode Validation
```typescript
// ✅ Valid
{ startTime: 1000, endTime: 2000, peakIntensity: 7, averageIntensity: 5 }

// ❌ Invalid - endTime before startTime
{ startTime: 2000, endTime: 1000 }
// Error: "End time must be after start time"

// ❌ Invalid - average > peak
{ peakIntensity: 5, averageIntensity: 7 }
// Error: "Average intensity cannot be greater than peak intensity"
```

### Medication Dose Validation
```typescript
// ✅ Valid - taken dose
{ amount: 50, status: 'taken', medicationId: '123', timestamp: 1000 }

// ✅ Valid - skipped dose with amount 0
{ amount: 0, status: 'skipped', medicationId: '123', timestamp: 1000 }

// ❌ Invalid - taken dose with amount 0
{ amount: 0, status: 'taken', medicationId: '123', timestamp: 1000 }
// Error: "Amount must be positive for taken doses"
// Database also rejects: CHECK(status != 'taken' OR amount > 0)
```

### Daily Status Validation
```typescript
// ✅ Valid - yellow day with status_type
{ date: '2025-01-10', status: 'yellow', statusType: 'prodrome' }

// ❌ Invalid - future date
{ date: '2030-01-01', status: 'green' }
// Error: "Date cannot be in the future"

// ❌ Invalid - statusType on non-yellow day
{ date: '2025-01-10', status: 'green', statusType: 'prodrome' }
// Error: "Status type can only be set for yellow days"
// Database also rejects: CHECK(status = 'yellow' OR status_type IS NULL)
```

## Multi-Layer Validation Strategy

This implementation provides defense in depth:

1. **Application Layer (Zod)**: Validates before database write, provides detailed error messages
2. **Database Layer (CHECK)**: Final enforcement, prevents invalid data even if application validation is bypassed
3. **TypeScript Types**: Compile-time type safety

## Testing
✅ **652/652 unit tests passing**  
✅ **15/15 E2E tests passing**  
✅ **TypeScript compilation successful** (0 errors)

## Technical Details

- **Zod version**: 3.25.76 (already installed as peer dependency)
- **Bundle size impact**: Minimal (~2kb gzipped)
- **Performance**: `safeParse()` has negligible overhead
- **Development logging**: Validation errors logged with `logger` utility (dev mode only)
- **Error messages**: Include field paths and constraints for easy debugging

## Migration Notes

The database schema version has been updated from 7 to 8. The migration will:
- Drop and recreate all tables with new CHECK constraints
- Preserve foreign key relationships and indexes
- Apps will automatically migrate on first launch after update

## Future Enhancements (Out of Scope)
As mentioned in issue #45, these could be addressed in future PRs:
- Form-level validation in UI components with real-time feedback
- Additional unit tests specifically for validation schema edge cases
- Validation error analytics/monitoring

Closes #45

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: app/src/schemas/dailyStatus.schema.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  'prodrome',
+  'postdrome',
+  'anxiety',
+  'other',
+]);
+
+// Date format validation (YYYY-MM-DD)
+const DateFormatRegex = /^\d{4}-\d{2}-\d{2}$/;
+
+export const DateStringSchema = z.string()
```
Comment (vfilby) [inline app/src/schemas/dailyStatus.schema.ts]: This is only used for the scheduled notifications where a timestamp doesn't make sense?

File: app/src/schemas/episode.schema.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+// Intensity value (0-10 scale)
+export const IntensityValueSchema = z.number()
+  .min(0, 'Intensity must be >= 0')
+  .max(10, 'Intensity must be <= 10');
+
+// Episode schema
+export const EpisodeSchema = z.object({
+  id: z.string().min(1, 'ID is required'),
+  startTime: z.number()
```
Comment (vfilby) [inline app/src/schemas/episode.schema.ts]: This applies in multiple places, would it be a good idea to define a 'timestamp' schema and reuse that rather than repeating the rules for each timestamp?
